janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java

```diff
@@ -74,7 +76,16 @@ protected static JanusGraphRelation findRelation(RelationIdentifier rId, JanusGr
             v = tmp;
             dir = Direction.IN;
         }
-        return ((VertexCentricQueryBuilder) v.query()).noPartitionRestriction().types((EdgeLabel) type).direction(dir).adjacent(other).edges();
+        VertexCentricQueryBuilder query =
+            ((VertexCentricQueryBuilder) v.query()).noPartitionRestriction().types(type).direction(dir).adjacent(other);
+
+        RelationType internalVertex = ((StandardJanusGraphTx) tx).getExistingRelationType(type.longId());
+        if (internalVertex instanceof InternalRelationType) {
+            if (((InternalRelationType) internalVertex).getConsistencyModifier() != ConsistencyModifier.FORK) {
```

Comment:
Would it be possible to add a test case that can fully cover this line? https://app.codecov.io/gh/JanusGraph/janusgraph/compare/2436/diff

Reply:
done